axlearn/common/input_lm_test.py

```diff
@@ -185,78 +134,169 @@ def test_training_lm_processor_single_example(self, text: str):
+                {
+                    "input_ids": [
+                        [1, 21820, 296, 2, 29],
+                        [3155, 1, 21820, 8114, 2],
+                    ],
+                    "target_labels": [
+                        [21820, 296, 2, 29, 3155],
+                        [1, 21820, 8114, 2, 29],
+                    ],
+                    "target_num_bytes": [18, 17],
+                },
+                {
+                    "input_ids": [
+                        [29, 3155, 1, 21820, 296],
+                        [2, 29, 3155, 0, 0],
+                    ],
+                    "target_labels": [
+                        [3155, 1, 21820, 296, 2],
+                        [29, 3155, 1, 0, 0],
+                    ],
+                    "target_num_bytes": [19, 3],
+                },
+            ],
+        ),
+    )
     @pytest.mark.skipif(
         not os.path.exists(t5_sentence_piece_vocab_file), reason="Missing testdata."
     )
-    def test_fake_text_lm_training_data(self):
-        expected_batches = [
-            {
-                "input_ids": [
-                    [1, 21820, 296, 2, 29, 3155],
-                    [29, 3155, 1, 21820, 1782, 2],
-                    [1782, 2, 29, 3155, 1, 21820],
-                ],
-                "target_labels": [
-                    [21820, 296, 2, 29, 3155, 1],
-                    [3155, 1, 21820, 1782, 2, 29],
-                    [2, 29, 3155, 1, 21820, 1712],
-                ],
-                "target_num_bytes": [19, 18, 18],
-            },
-            {
-                "input_ids": [
-                    [1, 21820, 296, 2, 29, 3155],
-                    [29, 3155, 1, 21820, 296, 2],
-                    [29, 3155, 1, 21820, 1712, 2],
-                ],
-                "target_labels": [
-                    [21820, 296, 2, 29, 3155, 1],
-                    [3155, 1, 21820, 296, 2, 29],
-                    [3155, 1, 21820, 1712, 2, 29],
-                ],
-                "target_num_bytes": [19, 20, 18],
-            },
+    def test_fake_text_lm_training_data(
+        self, packing_method: PackingMethodType, expected_batches: list[dict[str, Any]]
+    ):
+        texts = [
+            "hello world\n",
+            "hello moon\n",
         ]
-        self._test_fake_text_lm_training_data(
-            vocab_cfg=self.vocab_cfg, expected_batches=expected_batches
+
+        # window_size > len(texts) to repeat the sentence. 18 tokens in total.
+        # [    1,  21820,   296,     2,    29,  3155,     1, 21820,  8114,
+        #      2,     29,  3155,     1, 21820,   296,     2,    29,  3155]
+        window_size = 3
+
+        # Pad the concatenated sequence to 20 tokens.
+        # [    1,  21820,   296,     2,    29,  3155,     1, 21820,  8114,    2
+        #     29,  3155,     1, 21820,   296,     2,    29,  3155,      0,    0]
+        batch_size, max_len = 2, 5
+
+        # Always pad.
+        max_padding_fraction = 1.0
```

Comment:
With this change, do we still test the truncation behavior?

Reply:
Great catch! I added a test case of the truncation behavior. PTAL